sdc11073/mdib/devicemdib.py

```diff
@@ -1,981 +1,341 @@
 
-        self.sequenceId = uuid.uuid4().urn # this uuid identifies this mdib instance
-        
-        self._currentLocation = None  # or a SdcLocation instance
+        self._current_location = None  # or a SdcLocation instance
         self._annotators = {}
         self._current_transaction = None
 
-        self.preCommitHandler = None # preCommitHandler can modify transaction if needed before it is committed
-        self.postCommitHandler = None # postCommitHandler can modify mdib if needed after it is committed
+        self.pre_commit_handler = None  # pre_commit_handler can modify transaction if needed before it is committed
+        self.post_commit_handler = None  # post_commit_handler can modify mdib if needed after it is committed
         self._waveform_source = waveform_source or DefaultWaveformSource()
+        self._transaction_proc_cls = transaction_proc_cls
+        self._retrievability_episodic = []  # a list of handles
+        self.retrievability_periodic = defaultdict(list)
+        self.descriptor_factory = DescriptorFactory(self)
 
     @contextmanager
-    def mdibUpdateTransaction(self, setDeterminationTime=True):
-        #pylint: disable=protected-access
-        with self._trLock:
+    def transaction_manager(self, set_determination_time=True):
+        # pylint: disable=protected-access
+        with self._tr_lock:
             try:
-                self._current_transaction = _MdibUpdateTransaction(self)
+                self._current_transaction = MdibUpdateTransaction(self)
                 yield self._current_transaction
-                if callable(self.preCommitHandler):
-                    self.preCommitHandler(self, self._current_transaction) #pylint: disable=not-callable
+                if callable(self.pre_commit_handler):
+                    self.pre_commit_handler(self, self._current_transaction)  # pylint: disable=not-callable
                 if self._current_transaction._error:
-                    self._logger.info('mdibUpdateTransaction: transaction without updates!')
+                    self._logger.info('transaction_manager: transaction without updates!')
                 else:
-                    self._process_transaction(setDeterminationTime)
-                    if callable(self.postCommitHandler):
-                        self.postCommitHandler(self, self._current_transaction)  #pylint: disable=not-callable
+                    processor = self._transaction_proc_cls(self, self._current_transaction,
+                                                           set_determination_time, self._logger)
+                    processor.process_transaction()
+                    self._send_notifications(processor)
+                    self._current_transaction.mdib_version = self.mdib_version
+
+                    if callable(self.post_commit_handler):
+                        self.post_commit_handler(self, self._current_transaction)  # pylint: disable=not-callable

```

Comment:
Deleting context states is only one use case. I want to provide a universal mechanism for an application to react on changes in the mdib. E.g. a role provider could decide to start a follow up transaction because some rule/requirement demands it.
Agreed, there is a possibility to create inconsistencies, but that is in the responsibility of the application.